Add PtrConstruct template to use in Variant constructors

vnen opened the pull request:
Since the PtrToArg::encode requires the value to be constructed previously. With PtrConstruct this is not required.

vnen committed:
Add PtrConstruct template to use in Variant constructors

Since the PtrToArg::encode requires the value to be constructed
previously. With PtrConstruct this is not required.

akien-mga commented on core/variant/variant_construct.cpp:
```diff
@@ -39,6 +39,59 @@
+MAKE_PTRCONSTRUCT(int64_t);
+MAKE_PTRCONSTRUCT(double);
+MAKE_PTRCONSTRUCT(String);
+MAKE_PTRCONSTRUCT(Vector2);
+MAKE_PTRCONSTRUCT(Rect2);
+MAKE_PTRCONSTRUCT(Vector3);
+MAKE_PTRCONSTRUCT(Vector2i);
+MAKE_PTRCONSTRUCT(Rect2i);
+MAKE_PTRCONSTRUCT(Vector3i);
```
I suggest using the same order as in the `Variant::Type` enum, that makes it easier to parse the list and make sure nothing is missing.

vnen replied:
TBH I just copied the list from `method_ptrcall.h` but makes sense to keep the order.

akien-mga replied:
Might be worth changing in `method_ptrcall.h` at the same time then.

vnen replied:
Okay, I added a commit to change the other file too.

vnen committed:
Change template order in method_ptrcall.h

To be consistent with the enum in Variant so missing types can be more
easily spotted.

akien-mga commented:
Thanks!
